dj/sql/parse.py

```diff
@@ -2,26 +2,52 @@
 SQL parsing functions.
 """
 
-from typing import Optional
 
+from typing import TYPE_CHECKING
 
-def is_metric(query: Optional[str]) -> bool:
+from dj.errors import DJInvalidInputException
+
+if TYPE_CHECKING:
+    from dj.models.node import NodeRevision
+
+
+def check_is_metric(node: "NodeRevision"):
     """
-    Return if a SQL query defines a metric.
+    Check if a SQL query defines a metric.
 
-    The SQL query should have a single expression in its projections, and it should
-    be an aggregation function in order for it to be considered a metric.
+    The SQL query should have a single expression in its projections, it should
+    be an aggregation function and should have the same name
+    as the node in order for it to be considered a metric.
     """
 
     from dj.sql.parsing.backends.antlr4 import parse  # pylint: disable=C0415
 
-    if query is None:
-        return False
+    if node.query is None:
+        raise DJInvalidInputException("Invalid Metric. Node has no query.")
 
-    tree = parse(query)
+    tree = parse(node.query)
 
     # must have a single expression
     if len(tree.select.projection) != 1:
-        return False
-
-    return tree.select.projection[0].is_aggregation()  # type: ignore
+        raise DJInvalidInputException(
+            "Invalid Metric. Node query has does not have "
+            "a single expression in the projection.",
+        )
+
+    if (
+        not hasattr(tree.select.projection[0], "alias_or_name")
+        or tree.select.projection[0].alias_or_name.name != node.name  # type: ignore
+    ):
+        raise DJInvalidInputException(
+            "Invalid Metric. The expression in the projection "
```

Comment:
Instead of requiring that the expression has an alias that matches the metric node name, is there a way we can automatically set the alias for the user? I guess there could be some confusion if for example they define a metric `sales.num_orders` as `SELECT count(orders) AS total_order_count FROM ...`. If we replace the metric alias with `sales_DOT_num_orders` the user might be surprised because they were expecting it to be `total_order_count`. Maybe we should not allow any alias to be set by the user in the SQL definition. It would be a very reasonable explanation that the system defines metric expression aliases for consistency.

Reply:
How about this:
- if there is an alias and it does not match the expected alias, raise the error and tell the user
- otherwise if the alias is missing we can auto-add it

Reply:
I like that solution @agorajek, if the user already sets the proper alias it would definitely be unnecessary to tell them to remove it so that the system can re-add the exact same one 😅 .

@CircArgs that would mean this PR remains mostly as it is now, just have to break out the `not hasattr(tree.select.projection[0], "alias_or_name")` condition to its own if block and just set the alias for the user.

Reply:
Yeah, I suspect most users won't add aliases to the column when they write their metrics, so the addition of auto-adding aliases will be very useful!